src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -130,6 +132,38 @@ def __init__(self, msConfig, logger=None):
         self.wfCounters = {'cleaned': {},
                            'archived': {'normalArchived': 0,
                                         'forceArchived': 0}}
+        self.globalLocks = set()
+
+    def getGlobalLocks(self):
+        """
+        Fetches the list of 'globalLocks' from wmstats server and the list of
+        'parentLocks' from request manager. Stores/updates the unified set in
+        the 'globalLocks' instance variable. Returns the resultant unified set.
+        :return: A union set of the 'globalLocks' and the 'parentLocks' lists
+        """
+        globalLocks = set()
+        parentLocks = set()
+        self.logger.info("Fetching globalLocks list from wmstats server.")
+        try:
+            uri = 'globallocks'
+            globalLocks.update(set(self.wmstatsSvc._getResult(uri, clearCache=True, verb="GET")))
+        except Exception as ex:
+            msg = "Failed to refresh GlobalLocks list for the current poling cycle. "
+            msg += "Reusing the list (if any) from the previous run. Error: %s"
+            self.logger.warning(msg, str(ex))
+            globalLocks.update(self.globalLocks)
+        self.logger.info("Fetching parentLocks list from wmstats server.")
+        try:
+            uri = 'parentlocks/ALL_DOCS'
+            result = self.reqmgr2._getResult(uri, clearCache=True, verb="GET")
```

Comment:
Exactly the same comment goes here.

In addition to that, the URI is wrong. The REST endpoint is just `parentlocks`, e.g.:
https://cmsweb-testbed.cern.ch/reqmgr2/data/parentlocks

Reply:
Same reply, though. 
I tested it and strangely enough it works with both those URIs (with and without `/ALL_DOCS`). Nevertheless I am fixing it as you requested here.

Reply:
Same counter-reply here, I'd be very happy to see a proper method to fetch the list of parentlocks.

Reply:
Same as above.

Reply:
Done